rust/connlib/clients/apple/src/lib.rs

```diff
@@ -191,11 +193,19 @@ impl WrappedSession {
     ) -> Result<Self, String> {
         let secret = SecretString::from(token);
 
-        let session = Session::connect(
+        let (private_key, public_key) = keypair();
+        let login = LoginUrl::client(
             api_url.as_str(),
-            secret,
+            &secret,
             device_id,
             device_name_override,
+            public_key.to_bytes(),
+        )
+        .map_err(|e| e.to_string())?;
```

Comment:
Now that the `LoginUrl` type is separate, we might be able to further simplify this based on which client calls it.